src/hotspot/cpu/riscv/gc/shared/barrierSetAssembler_riscv.cpp

```diff
@@ -179,11 +179,13 @@ void BarrierSetAssembler::eden_allocate(MacroAssembler* masm, Register obj,
     int32_t offset = 0;
     __ bind(retry);
 
+    Register tmp = t0;
```

Comment:
No need to introduce local variable "tmp" here. I think we can use t0 directly here.

Reply:
This is a reasonable point -- changed as proposal, passed fastdebug build, and tested a simple `java -XX:+UseSerialGC -XX:-UseTLAB -XX:TieredStopAtLevel=1`.